src/docfx/lib/ProcessUtility.cs

```diff
@@ -121,20 +119,34 @@ public static bool IsNotFound(Win32Exception ex)
                    ex.ErrorCode == 2; // ERROR_FILE_NOT_FOUND = 0x2, The system cannot find the file specified
         }
 
-        private static async Task<FileStream> AcquireFileMutex(string lockPath, int retry, TimeSpan retryTimeSpanInterval)
+        private static async Task<FileStream> AcquireFileMutex(string mutexName, string lockPath)
         {
-            var retryCount = 0;
+            var retryDelay = 100;
+            var lastWait = DateTime.UtcNow;
+
             while (true)
```

Comment:
>true [](start = 19, length = 4)

retry forever?

Reply:
Yes. I tend to let build server having a total `docfx` timeout like an hour. It helps with all the other cases where we might stuck, like a code bug that causes deadlock. 

When you use other lock mechanism, like `lock (obj)`, there is no timeout by default.

---
In reply to: [216057373](https://github.com/dotnet/docfx/pull/3384#discussion_r216057373) [](ancestors = 216057373)

Reply:
What about keep timeout as a parameter (default to forever) in utility layer, and let business layer to decide the time out? 

---
In reply to: [216180485](https://github.com/dotnet/docfx/pull/3384#discussion_r216180485) [](ancestors = 216180485,216057373)